src/commands/protocols/common/epoch-check/blockchain-epoch-check-command.js

```diff
@@ -188,6 +190,37 @@ class BlockchainEpochCheckCommand extends Command {
                         blockchainAgreementData.proofWindowOffsetPerc;
                 }
 
+                const assertionId = await this.blockchainModuleManager.getAssertionIdByIndex(
+                    serviceAgreement.blockchainId,
+                    serviceAgreement.assetStorageContractAddress,
+                    serviceAgreement.tokenId,
+                    0,
+                );
+
+                const newServiceAgreementId = this.serviceAgreementService.generateId(
+                    serviceAgreement.blockchainId,
+                    serviceAgreement.assetStorageContractAddress,
+                    serviceAgreement.tokenId,
+                    assertionId,
+                    CONTENT_ASSET_HASH_FUNCTION_ID, // 1 - sha256
+                );
+
+                if (serviceAgreement.agreementId !== newServiceAgreementId) {
+                    const newKeyword = this.blockchainModuleManager.encodePacked(
+                        serviceAgreement.blockchainId,
+                        ['address', 'bytes32'],
+                        [serviceAgreement.assetStorageContractAddress, assertionId],
+                    );
+
+                    await this.updateServiceAgreementForTokenId(
```

Comment:
We can just do this in the previous update function, no need to check even if it's correct or not